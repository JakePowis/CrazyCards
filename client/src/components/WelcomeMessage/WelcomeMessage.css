.welcomeMessage {

    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    height: 100%;

}

.cardIcon {
    padding: 6rem;
    max-height: 300px;
    width: auto;
}



@media only screen and (max-width: 600px) {
    .welcomeMessage {
     text-align: center;
    }

    .cardIcon {
        padding: 4rem;
        max-height: 180px;
        width: auto;
    }
  }