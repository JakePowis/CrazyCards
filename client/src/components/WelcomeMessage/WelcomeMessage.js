import React from 'react'
import StepCounter from '../StepCounter/StepCounter'
import './WelcomeMessage.css'
import cardIcon from './cardicon.png'

export default function WelcomeMessage({toggleView, setToggleView}) {
    return (
        <div className="welcomeMessage">
        <StepCounter step1 />
        <div>Welcome to Crazy Cards. </div>
        <div>Enter some quick details to match you with a card:</div>
        <img src={cardIcon} className="cardIcon" alt="image of credit cards"/>
        <button onClick={()=> setToggleView("form")}>Find Me a Card!</button>
        </div>
        
    )
}
