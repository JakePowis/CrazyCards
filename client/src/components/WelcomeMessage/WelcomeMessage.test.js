import React from 'react';
import ReactDom from 'react-dom';
import WelcomeMessage from './WelcomeMessage.js'
import { render, cleanup } from '@testing-library/react';

import renderer from "react-test-renderer"

afterEach(cleanup)

test("Welcome Message Renders Correctly", ()=> {
    const div = document.createElement("div");
    ReactDom.render(<WelcomeMessage/>, div)
})

test('Renders page title', () => {
    const { getByText } = render(<WelcomeMessage/>);
    const divElement = getByText(/Welcome to Crazy Cards/i);
    expect(divElement).toBeInTheDocument();
  });

  test("matches snapshot", ()=> {

    const tree = renderer.create(<WelcomeMessage/>).toJSON()
    expect(tree).toMatchSnapshot();
})
  