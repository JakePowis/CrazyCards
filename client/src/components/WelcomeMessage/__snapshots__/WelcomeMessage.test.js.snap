// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="welcomeMessage"
>
  <div
    className="steps"
  >
    <div
      className="active"
    >
      Introduction
    </div>
    <div
      className=""
    >
      Enter Details
    </div>
    <div
      className=""
    >
      Select Card
    </div>
  </div>
  <div>
    <h1>
      Welcome to Crazy Cards 
    </h1>
  </div>
  <div>
    Enter some quick details and we'll match 
    <b>
      you
    </b>
     with your perfect card
  </div>
  <img
    alt=""
    className="cardIcon"
    src="cardicon.png"
  />
  <button
    onClick={[Function]}
  >
    Find My Perfect Card
  </button>
</div>
`;
